pallene/translator.lua

```diff
@@ -5,6 +5,24 @@
 
 local util = require "pallene.util"
 
+--
+-- This file implements a Pallene to Lua translator.
+--
+-- The Pallene compiler is divided into two logical ends:
+-- * The frontend which parses Pallene source code to generate AST and performs semantic analysis.
+-- * The backend which generates C source code.
+--
+-- Both these ends are decoupled, this provides us with the flexibility to integrate another backend
+-- that generates Lua. The users can run the compiler with `--emit-lua` trigger the translator to
+-- generate plain Lua instead of C.
+--
+-- The generation of Lua is performed by a different backend (implemented here). It accepts input
+-- string and the AST generated by the parser. The generator then walks over the AST to replacing
+-- type annotations with white space. Interestingly spaces, newlines, comments and pretty much
+-- everything else other than type annotations are retained in the translated code. Thus, the
+-- formatting in the original input is preserved.
```

Comment:
Good explanation. Could you please also mention that the line numbers are preserved, so the error messages point to the right place?